hostinfo/hostname_macos.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build cgo && darwin && !ios
+
+package hostinfo
+
+/*
+#cgo CFLAGS: -x objective-c
+#cgo LDFLAGS: -framework Foundation
+#import <Foundation/Foundation.h>
+
+const char *getHostname() {
+	NSString *hostname = [[NSHost currentHost] localizedName];
+    if (hostname != nil) {
+        const char *hostnameCString = [hostname UTF8String];
+        if (hostnameCString != NULL) {
+            return strdup(hostnameCString);
+        }
+    }
+    return NULL;
+}
+*/
+import "C"
+
```

Comment:
Ideally we'd limit the spread of cgo infection and do the setter thing we do for Android and stuff like hostinfo.SetDeviceModel ... then have the ipn-go-bridge Swift/C code set it before calling into the Go entrypoint.

Reply:
How is that going to work on open-source tailscaled?

Reply:
Can open source tailscaled shell out to the "hostname" binary? Does that still work?

Reply:
No, it also appears to return `Mac.home`.

Reply:
They broke the built-in `hostname` binary !?! Uhhhhhhhhh